libsel4vmmplatsupport/src/arch/arm/smc.c

```diff
@@ -45,9 +46,15 @@ static uintptr_t smc_get_function_number(uintptr_t func_id)
 }
 
 int handle_smc(vm_vcpu_t *vcpu, uint32_t hsr)
+{
+    return vcpu->vm->arch.vm_smc_handler(vcpu);
+}
```

Comment:
Perhaps move all the `vm_get_thread_context`, `vm_set_thread_context` and `advance_vcpu_fault` calls here and let it pass the user context to the callback function.

This avoids code duplication and makes it easier for users to write a callback function. This would need a slight change to `handle_psci` too.

Reply:
Moved these into `handle_smc`.